### ISSUE TITLE
Allow passing providers to PlaygroundModule

### ISSUE THREAD

**Issue by cgatian** [SoCreate/angular-playground#32]
Angular allows you to override the `ErrorHandler` service with your own implementation.  However, only the [app root](https://github.com/angular/angular/blob/71f5b73296708014b740fb5dd0145c0599de7a19/packages/core/src/application_ref.ts#L297) can provide this override.  Since PlaygroundModule is the `app root`, it's not possible to provide an override for the `ErrorHandler` service.

I believe the only workaround is to provide a static method on PlaygroundModule that allows you to include an optional array of providers.

**Comment by jschwarty** [comment]
This kinda falls under the same case as #28. Looks like Playground needs this for multiple reasons. :) Will think about a way to approach this...

**Comment by cgatian** [comment]
Sounds good. If you come up with a plan, post her here. I don't mind making the PR.

**Comment by cgatian** [comment]
In the short term, couldnt we pass providers to the module using a static method?
```
PlaygroundModule.withProviders([
    { provide: ErrorHandler, useClass: CustomErrorHandler }
])
```

**Comment by jschwarty** [comment]
Yeah that might be an approach.

**Comment by jschwarty** [comment]
@cgatian If you have any thoughts on a way to pull this off it'd be cool to see. :) The goal would be to try and either leverage the existing config or file stuff and not have to have the user add another file to their project to get this included. But that may end up being the only way.

Here are the goals I think would be good to hit:
- Keep it in code (imports and typings can be used rather than say an array of type string names in the json config file).
- Could it be done either in the user's playground `main.ts` or maybe in the `initialize` api method?
- Or maybe it needs to be handled by the playground cli, and if so maybe the `SandboxConfig` can get a `forRoot` property that can collect any root NgModules that the cli can aggregate through and write out to the sandboxes.ts file for use by the `PlaygroundModule`.

**Comment by jschwarty** [comment]
Fixed in PR #36 and will be available in upcoming release 1.7.0

The core Playground stuff got moved into a NgModule named `PlaygroundCommonModule` that can be imported and used in a user's own custom module you create in their project, allowing them to set other imports, providers, etc at the root NgModule level as needed. The `AppComponent` for Playground was added to the exports as well so it can be bootstrapped from that custom root NgModule.

The angularplayground.it site will get updated with docs to show this implementation option.

**Comment by cgatian** [comment]
Thanks @jschwarty. I like the approach you ended up with. Much better than a static function.
